packages/core/src/external/quest/client.ts

```diff
@@ -43,11 +44,65 @@ export class QuestSftpClient extends SftpClient {
     }
   }
 
+  /**
+   * Downloads all Quest responses from the Quest SFTP server. If this client is configured with a
+   * replica, it will only download files that have not already been downloaded to S3.
+   */
+  async downloadAllResponses(): Promise<QuestResponseFile[]> {
+    const responseFileNamesInReplica = await this.listResponseFileNamesFromReplica();
+    const alreadyDownloadedFileNames = new Set(responseFileNamesInReplica);
+
+    try {
+      await this.connect();
+      const fileNames = await this.listResponseFileNamesFromQuest();
+      this.log(`Found ${fileNames.length} file updates in Quest SFTP directory`);
+
+      const responseFiles: QuestResponseFile[] = [];
+      for (const fileName of fileNames) {
+        if (alreadyDownloadedFileNames.has(fileName)) continue;
+        this.log(`Downloading ${fileName}...`);
+        const fileContent = await this.readFromQuest(fileName);
+        responseFiles.push({ fileName, fileContent });
+        this.log(`Finished downloading ${fileName}`);
+      }
+      return responseFiles;
+    } catch (error) {
+      throw new MetriportError(`Failed to download Quest responses`, error, {
+        context: "QuestSftpClient",
+      });
+    } finally {
+      await this.disconnect();
+    }
+  }
+
+  /**
+   * Lists all response file names from the Quest replica.
+   */
+  private async listResponseFileNamesFromReplica(): Promise<string[]> {
+    if (!this.replica) {
+      return [];
+    }
+    const incomingDirectoryReplica = this.replica.getReplicaPath(this.incomingDirectory);
```

Comment:
The base implementation just removes the leading `/` character from the SFTP file name, so it is a valid S3 key.